### PR TITLE
[docs] Mobile doc updates

### PR DESCRIPTION
**Pull request by edgchen1**
### Description

- Update Android build instructions.
- Remove references to "mobile" packages with reduced operators and minimal build which have been removed in 1.19.
  In particular, these packages:
  - onnxruntime-mobile-c/objc (iOS)
  - onnxruntime-mobile (Android)

https://edgchen1.github.io/onnxruntime/docs/

### Motivation and Context

Update and remove obsolete documentation.

### CODE REVIEW

File: docs/tutorials/mobile/index.md [hunk 1]
```diff
@@ -80,36 +80,10 @@ Another way of reducing the model size is to find a new model with the same inpu
...
-|Architecture|Package|Size|
-|-|-|-|
-|arm64|onnxruntime-android|12.2 MB|
-||onnxruntime-mobile|3.2 MB|
-|arm32|onnxruntime-android|8.4 MB|
-||onnxruntime-mobile|2.3 MB|
-||custom (MobileNet)|_Coming soon_|
-
-The iOS package is a static framework and so the library package size is not a good indication of the actual contribution to the application binary size. The above sizes for Android are good estimates for iOS.
+TODO compare full package vs custom build binary size
```
Comment (edgchen1) [inline docs/tutorials/mobile/index.md:89]: get some measurements of Android package size

File: docs/build/android.md [hunk 2]
```diff
@@ -81,9 +81,8 @@ Resources:
 * Install the NDK
   * Find the available NDK versions by running `sdkmanager --list`
   * Install
-    * you can install a specific version or the latest (called 'ndk-bundle') e.g. `sdkmanager --install "ndk;21.1.6352462"`
+    * install the desired version, e.g., `sdkmanager --install "ndk;21.1.6352462"`
     * NDK path in our example with this install would be `.../Android/ndk/21.1.6352462`
-    * NOTE: If you install the ndk-bundle package the path will be `.../Android/ndk-bundle` as there's no version number
```
Comment (edgchen1) [inline docs/build/android.md:86]: note: ndk-bundle is obsolete

File: docs/tutorials/mobile/deploy-ios.md [hunk 3]
```diff
@@ -88,7 +88,7 @@ Here's an example screenshot of the app:
    pod install
    ```
 
-   The `Podfile` contains the `onnxruntime-mobile-objc` dependency, which is the pod containing the Objective C API.
+   The `Podfile` contains the `onnxruntime-objc` dependency, which is the pod containing the Objective C API.
```
Comment (edgchen1) [inline docs/tutorials/mobile/deploy-ios.md:91]: need to update onnxruntime-inference-examples

**Review by sheetalarkadam — DISMISSED** [review]

**Review by vraspar — DISMISSED** [review]

File: docs/tutorials/mobile/index.md [hunk 4]
```diff
@@ -80,36 +80,18 @@ Another way of reducing the model size is to find a new model with the same inpu
...
-   * Android Java/C/C++: onnxruntime-mobile
-   * iOS C/C++: onnxruntime-mobile-c
-   * iOS Objective-C: onnxruntime-mobile-objc
+One of the outputs of the ORT format conversion is a build configuration file, containing a list of operators from your model(s) and their types. You can use this configuration file as input to the custom runtime binary build.
 
-   These mobile packages have a smaller binary size but limited feature support, like a reduced set of operator implementations and the model must be converted to [ORT format](../../performance/model-optimizations/ort-format-models.md#convert-onnx-models-to-ort-format).
+To give an idea of the binary size difference between the pre-built package and a custom build:
 
-   See the [install guide](../../install/#install-on-web-and-mobile) for package specific instructions.
+File|Pre-built package size (bytes)|Custom build size (bytes)
```
Comment (edgchen1) [inline docs/tutorials/mobile/index.md:91]: mention version 1.18.0

File: docs/install/index.md [hunk 5]
```diff
@@ -438,9 +417,9 @@ below:
...
-| Objective-C  | CocoaPods: **onnxruntime-mobile-objc**                                                                                                            |                                                                                                                                               | [View](../install/index.md#install-on-ios)                                                     |
+| Android      | [**com.microsoft.onnxruntime:onnxruntime-android**](https://search.maven.org/artifact/com.microsoft.onnxruntime/onnxruntime-android)                |                                                                                                                                               | [View](../install/index.md#install-on-android)                                                     |
```
Comment (edgchen1) [inline docs/install/index.md:420]: fix indentation of `|`

File: docs/install/index.md [hunk 6]
```diff
@@ -438,9 +417,9 @@ below:
...
+| iOS (C/C++)  | CocoaPods: **onnxruntime-c**                                                                                                               |                                                                                                                                               | [View](../install/index.md#install-on-ios)                                                     |
+| Objective-C  | CocoaPods: **onnxruntime-objc**                                                                                                            |                                                                                                                                               | [View](../install/index.md#install-on-ios)                                                     |
```
Comment (edgchen1) [inline docs/install/index.md:422]:
```suggestion
| Android      | [**com.microsoft.onnxruntime:onnxruntime-android**](https://search.maven.org/artifact/com.microsoft.onnxruntime/onnxruntime-android)              |                                                                                                                                               | [View](../install/index.md#install-on-android)                                                    |
| iOS (C/C++)  | CocoaPods: **onnxruntime-c**                                                                                                                      |                                                                                                                                               | [View](../install/index.md#install-on-ios)                                                     |
| Objective-C  | CocoaPods: **onnxruntime-objc**                                                                                                                   |                                                                                                                                               | [View](../install/index.md#install-on-ios)                                                     |
```

File: docs/install/index.md [hunk 7]
```diff
@@ -438,9 +417,9 @@ below:
...
-| Objective-C  | CocoaPods: **onnxruntime-mobile-objc**                                                                                                            |                                                                                                                                               | [View](../install/index.md#install-on-ios)                                                     |
+| Android      | [**com.microsoft.onnxruntime:onnxruntime-android**](https://search.maven.org/artifact/com.microsoft.onnxruntime/onnxruntime-android)              |                                                                                                                                               | [View](../install/index.md#install-on-android)                                                    |
```
Comment (edgchen1) [inline docs/install/index.md:420]:
```suggestion
| Android      | [**com.microsoft.onnxruntime:onnxruntime-android**](https://search.maven.org/artifact/com.microsoft.onnxruntime/onnxruntime-android)              |                                                                                                                                               | [View](../install/index.md#install-on-android)                                                 |
```

**Review by skottmckay — APPROVED** [review]

**Pull request merged**